crates/router/src/connector/cybersource/transformers.rs

```diff
@@ -726,6 +744,76 @@ impl
                         }),
                     )
                 }
+                Some(payments::MandateReferenceId::NetworkTokenWithNTI(mandate_data)) => {
+                    let (original_amount, original_currency) = match network
+                        .clone()
+                        .map(|network| network.to_lowercase())
+                        .as_deref()
+                    {
+                        //This is to make original_authorized_amount mandatory for discover card networks in NetworkMandateId flow
+                        Some("004") => {
```

Comment:
In hard coding the strings, create constants in the file and use,

The naming of variable, should explain why we hard-coding it and constant should have comments explain it.

Reply:
will create an issue for this, will take it up in separate pr